http/src/request/guild/sticker/create_guild_sticker.rs

```diff
@@ -108,21 +111,97 @@ impl<'a> AuditLogReason<'a> for CreateGuildSticker<'a> {
             guild_id: self.guild_id.get(),
         });
 
-        let mut form = Form::new();
-
-        form.part("name".as_bytes(), self.fields.name.as_bytes());
-
-        form.part("description".as_bytes(), self.fields.description.as_bytes());
-
-        form.part("tags".as_bytes(), self.fields.tags.as_bytes());
-
-        form.part("file".as_bytes(), self.fields.file);
-
-        request = request.form(form);
+        let boundary = multipart::random_boundary();
+
+        let mut capacity = 0;
+
+        capacity += BOUNDARY_TERMINATOR.len() * 6;
+        capacity += boundary.len() * 5;
+        capacity += KEY_PRE.len() * 4;
+        capacity += KEY_POST.len() * 4;
+        capacity += NEWLINE.len() * 12;
+
+        capacity += NAME.len();
+        capacity += self.fields.name.len();
+
+        capacity += DESCRIPTION.len();
+        capacity += self.fields.description.len();
+
+        capacity += TAGS.len();
+        capacity += self.fields.tags.len();
+
+        capacity += FILE.len();
+        capacity += self.fields.file.len();
+
+        let mut buffer = Vec::with_capacity(capacity);
+
+        // Write the first boundary.
+        buffer.extend(BOUNDARY_TERMINATOR);
+        buffer.extend(&boundary);
+        buffer.extend(NEWLINE);
+
+        // Write the name to the form.
+        buffer.extend(KEY_PRE);
+        buffer.extend(NAME);
+        buffer.extend(KEY_POST);
+        buffer.extend(NEWLINE);
+        buffer.extend(NEWLINE);
+        buffer.extend(self.fields.name.as_bytes());
+        buffer.extend(NEWLINE);
+        buffer.extend(BOUNDARY_TERMINATOR);
+        buffer.extend(&boundary);
+        buffer.extend(NEWLINE);
+
+        // Write the description to the form.
+        buffer.extend(KEY_PRE);
+        buffer.extend(DESCRIPTION);
+        buffer.extend(KEY_POST);
+        buffer.extend(NEWLINE);
+        buffer.extend(NEWLINE);
+        buffer.extend(self.fields.description.as_bytes());
+        buffer.extend(NEWLINE);
+        buffer.extend(BOUNDARY_TERMINATOR);
+        buffer.extend(&boundary);
+        buffer.extend(NEWLINE);
+
+        // Write the tags to the form.
+        buffer.extend(KEY_PRE);
+        buffer.extend(TAGS);
+        buffer.extend(KEY_POST);
+        buffer.extend(NEWLINE);
+        buffer.extend(NEWLINE);
+        buffer.extend(self.fields.tags.as_bytes());
+        buffer.extend(NEWLINE);
+        buffer.extend(BOUNDARY_TERMINATOR);
+        buffer.extend(&boundary);
+        buffer.extend(NEWLINE);
+
+        // Write the file to the form.
+        buffer.extend(KEY_PRE);
+        buffer.extend(FILE);
+        buffer.extend(KEY_POST);
+        buffer.extend(NEWLINE);
+        buffer.extend(NEWLINE);
+        buffer.extend(self.fields.file);
+        buffer.extend(NEWLINE);
+        buffer.extend(BOUNDARY_TERMINATOR);
+        buffer.extend(&boundary);
+        buffer.extend(NEWLINE);
+
+        request = request.form(Form::new(boundary, buffer));
```

Comment:
I am not a big fan of replicating logic here, though I can see why it is.

Reply:
I am open to adding a library that simply builds multipart forms for us, to save mistakes on my part